pylint/testutils/output_line.py

```diff
@@ -16,19 +19,30 @@ class MessageTest(
 ):
     """Used to test messages produced by pylint. Class name cannot start with Test as pytest doesn't allow constructors in test classes."""
 
-    def __new__(cls, msg_id, line=None, node=None, args=None, confidence=None):
+    def __new__(
+        cls,
+        msg_id: str,
+        line: Optional[int] = None,
+        node: Optional[nodes.NodeNG] = None,
+        args: Any = None,
```

Comment:
Similar to what we did for `Message` in #5050. This should at some point be fully investigated and changed.